### ISSUE TITLE
Proposal: Flag use of identifier in prop value when not in scope

### ISSUE THREAD

**Issue by DrewML** [yannickcr/eslint-plugin-react#646]
Note: I could not find an applicable rule that already exists in the readme. If I missed it, please let me know, and I apologize for the extra noise.

----

This one bit me pretty bad today.

Example:
```javascript
<ElementName someProp={status} />
```

In this ES module, there was no defined identifier for `status`. However, [there is a global for `status`](http://www.w3schools.com/jsref/prop_win_status.asp) in browsers, so our unit tests passed when run in a browser. We only noticed the issue when starting to run tests in node.

Is there any interest in adding this rule? I'd be happy to contribute it. I started an (ugly) rough draft of it [here](http://astexplorer.net/#/0snxAVn440)

**Comment by ljharb** [comment]
Does the normal eslint rule for undefined vars not catch this?

**Comment by DrewML** [comment]
Huh - interesting. It's not flagging this particular scenario in our codebase, but I at least verified in ASTExplorer that it _should_ be working. For some reason I was operating under the assumption that ESLint rules wouldn't be checking JSX nodes in the AST, but I guess they're just looking for _any_ identifier.

Thanks for the sanity check @ljharb.

**Comment by DrewML** [comment]
Should have realized sooner - wasn't being flagged because `browser` was set to `true` for `env`, and [`status` is part of the browser globals](https://github.com/sindresorhus/globals/blob/master/globals.json#L550)

**Comment by ljharb** [comment]
Can't really avoid that, I think - if `browser` is `true`, then all browser globals are bugs waiting to happen :-/
